### ISSUE TITLE
Forward incoming UDP to parent SOCK5 proxy 

### ISSUE THREAD

**Issue by john-d13** [nadoo/glider#235]
As the subject reads,  is gider capable of forwarding incoming UDP packets to an upstream parent Socks5 proxy?  In other worlds proxy protocol chaining.

I see in your documentation it suggests you do but there are no config samples. Would it be possible to provide a reference?

**Comment by nadoo** [comment]
Did you mean forwarding udp packets from local socks5 server to upstream socks5 server? yes.

As there's not so much software that forwards udp via socks5 server, we can use glider and dig to test it:

1. setup an upstream socks5 server:
```
glider -verbose -listen socks5://127.0.0.1:1080
```

2. forward local udp port 53 to 8.8.8.8:53 via upstream socks5 server:
```
glider -verbose -listen udp://:53 -forward "socks5://127.0.0.1:1080,udp://8.8.8.8:53"
```

3. test local udp port 53:
```
dig @127.0.0.1 www.apple.com
```

You can see the log or capture packets by using Wireshark to see what happened.

**Comment by john-d13** [comment]
Thank you for the quick reply nadoo!

Here is the flow,  let me know if possible and I can do some testing

IPTables PREROUTING -p udp -s 0/0 --dports 12345

IPTABLES -> udp://0.0.0.0:12345 (glider listener) -> socks5://{external IP:3345}

So the local listener will forward to external socks5 proxy and back.  Its not DNS traffic,  its actual payload traffic for my gameserver.

**Comment by emptyteeth** [comment]
> Thank you for the quick reply nadoo!
> 
> Here is the flow, let me know if possible and I can do some testing
> 
> IPTables PREROUTING -p udp -s 0/0 --dports 12345
> 
> IPTABLES -> udp://0.0.0.0:12345 (glider listener) -> socks5://{external IP:3345}
> 
> So the local listener will forward to external socks5 proxy and back. Its not DNS traffic, its actual payload traffic for my gameserver.

I think nadoo's reply has covered your needs already, It could be any udp traffic, not only dns

```shell
glider -verbose -listen udp://:12345 -forward "socks5://{upstream socks5 addr}:{upstream sock5 port},udp://{gameserver addr}:{gameserver port}"
```

**Comment by john-d13** [comment]
Yes but the destination UDP IP:PORT (gameserver address:port) is unknown. The request will be received dybamically from the iptables prerouting rules and redirected to local udp port mapper (udp://:12345)

**Comment by emptyteeth** [comment]
> Yes but the destination UDP IP:PORT (gameserver address:port) is unknown. The request will be received dybamically from the iptables prerouting rules and redirected to local udp port mapper (udp://:12345)

Well, one method I know of is `TPROXY` udp proxying, which not implemented in glider, here's an [example](https://github.com/v2fly/v2fly-github-io/blob/master/docs/config/protocols/dokodemo.md#%E9%80%8F%E6%98%8E%E4%BB%A3%E7%90%86%E9%85%8D%E7%BD%AE%E6%A0%B7%E4%BE%8B) from v2ray

**Comment by nadoo** [comment]
Closed as tproxy already supported in v0.15.0+
